projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -1281,9 +1349,15 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
         return DatePickerUtil.addPromptCharsEditMode(this.dateFormatParts, this.value, changedValue);
     }
 
-    private _onTouchedCallback: () => void = () => { };
-
-    private _onChangeCallback: (_: Date) => void = () => { };
+    public _updateValidity() {
+        this._onTouchedCallback();
+        const input = this.readonlyInputDirective || this.editableInputDirective || this.input;
```

Comment:
So the required visual is applied to the inputs from the view only, but the validity also affects the content input? Kind of on the fence about manipulating the external one.

Reply:
kind of yes :)